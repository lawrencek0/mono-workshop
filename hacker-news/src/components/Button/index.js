import React from 'react';
import './index.css';

const Button = ({onClick, className = '', children}) =>
  <button
    className={className}
    onClick={onClick}
    type="button"
  >
    {children}
  </button>

export default Button;