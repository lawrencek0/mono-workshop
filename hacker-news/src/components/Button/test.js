import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('Button', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Give me more</Button>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Button>Give me more</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});