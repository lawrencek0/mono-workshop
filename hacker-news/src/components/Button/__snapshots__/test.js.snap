exports[`Button snapshots 1`] = `
<button
  className=""
  onClick={undefined}
  type="button">
  Give me more
</button>
`;
